### PR TITLE
Add code of conduct document

### PR DESCRIPTION
**Pull request by imurchie**
The Appium project, going forward, needs to define its processes. Toward that end, we need a code of conduct to delineate what is and is not acceptable.

### CODE REVIEW

**Comment by jlipps** [comment]
I'm in favor of this change. Going forward, Appium needs to mature in certain process, organization, and governance areas in order to be a long-term self-sustaining community. I think this is an important step in maintaining the helpful community we've built our reputation around already.

pinging @bootstraponline @penguinho @Jonahss @moizjv @sebv @scottdixon

**Comment by bootstraponline** [comment]
I like the [Ubuntu Code of Conduct](http://www.ubuntu.com/about/about-ubuntu/conduct). I'm not a fan of this one. :-1:

**Comment by jlipps** [comment]
@bootstraponline can you highlight the specific aspects of this proposed CoC that you're not a fan of?

File: coc.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+This code of conduct outlines our expectations for all those who participate in our community, as well as the consequences for unacceptable behavior.
+
+We expect all members of the Appium community, including paid and unpaid agents, administrators, and users, to abide by this Code of Conduct at all times in all Appium community venues, online and in person, and in one-on-one communications pertaining to the Appium community.
+
+
+### Appium Community Citizenship
+
+A supplemental goal of this Code of Conduct is to increase community citizenship by encouraging participants to recognize and strengthen the relationships between our actions and their effects on our community. Communities mirror the societies in which they exist, and positive action is essential to counteract the many forms of inequality and abuses of power that exist in society. Unfortunately, many of today's societies come laden with assumptions about persons or personal attributes that are harmful or disprespectful. As a global community, we want Appium to represent the _best_ of how we can respect each other, and to that end we desire all community members to participate in creating a safe and creative space.
+
```
Comment (moizjv) [inline coc.md]: :100: :+1:

**Comment by jlipps** [comment]
@bootstraponline @imurchie made another pass, thoughts?

**Comment by imurchie** [comment]
I like this iteration.

**Comment by jlipps** [comment]
ok, made a few final tweaks. @imurchie @bootstraponline @moizjv @scottdixon @Jonahss @sebv @penguinho looking for a +1 and then we'll call it done

File: coc.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+* Sharing sexual images or using sexually explicit language
+
+In general: treat others how you would like to be treated, were you in their place. Don't be a jerk. _Do_ ask questions. _Do_ keep conflicts productively focused on technical issues. _Do_ think before you speak; remember that what is perceived as a funny witticism in your group of friends might be hurtful or reinforce hurtful stereotypes in the context of our diverse online community. _Do_ remember that we are all people, not robots, and all equally deserving of sensitivity and respect. (If and when robots join our community, let's treat them with respect too!)
+
+
+### What will organizers do about inappropriate behavior?
+
+If we notice you doing or saying something appropriate, an organizer will explain why it's inappropriate and ask you to stop. We won't demonize or vilify you. But please do stop the inappropriate behavior so we can get back to writing and discussing code in a safe environment. If you have philosophical disagreements about what's actually inappropriate, please take them to a separate public or private conversation with an Appium maintainer so we don't turn pull requests into an ethics debate.
```
Comment (Jonahss) [inline coc.md]: a very tiny typo radically changes the meaning of this sentence.
"If we notice you doing or saying something appropriate"

(should be "inappropriate")

**Comment by imurchie** [comment]
:+1:

File: coc.md (hunk 2 above)
Comment (imurchie) [inline coc.md]: Haha! Good eye.

Comment (moizjv) [inline coc.md]: haha

**Comment by Jonahss** [comment]
Other than typo above, :+1:

**Comment by scottdixon** [comment]
:+1:

**Comment by jlipps** [comment]
typo fixed! thanks for that.

**Comment by bootstraponline** [comment]
:+1:

**Comment by moizjv** [comment]
:+1:

**Comment by jlipps** [comment]
changed page name and linked from contributing doc as well. merging, thanks for the feedback everyone!

**Pull request merged**